language-plutus-core/generators/Language/PlutusCore/Generators/Interesting.hs

```diff
@@ -48,17 +48,17 @@ genOverapplication = do
     TermOf ti2 i2 <- genTypedBuiltinSmall typedInt1
     TermOf tj1 j1 <- genTypedBuiltinSmall typedInt2
     TermOf tj2 j2 <- genTypedBuiltinSmall typedInt2
-    let term =
-            mkIterApp ()
-                (TyInst ()
-                    (mkIterApp ()
-                        (TyInst ()
-                            (builtinNameAsTerm LessThanInteger)
-                            (TyInt () s1))
-                        [ti1, ti2])
-                    int2)
-                [tj1, tj2]
-        value = TypedBuiltinValue typedInt2 $ if i1 < i2 then j1 else j2
+    term <- rename $
```

Comment:
Could you stick a `rename` in where you actually run the `TermGen` in the end, or does that not work?

Reply:
There are several functions that run `TermGen`s.

I think, it's best for things in `Quote` to have unique names, otherwise it's too easy to make a mistake.